source/adapters/level_zero/event.cpp

```diff
@@ -1500,7 +1500,20 @@ ur_result_t _ur_ze_event_list_t::createAndRetainUrZeEventList(
 
         std::shared_lock<ur_shared_mutex> Lock(EventList[I]->Mutex);
 
-        if (Queue && Queue->Device != CurQueueDevice &&
+        ur_device_handle_t QueueRootDevice;
+        ur_device_handle_t CurrentQueueRootDevice;
+        if (Queue) {
+          QueueRootDevice = Queue->Device;
+          CurrentQueueRootDevice = CurQueue->Device;
```

Comment:
This section of the code may not be holding the lock on `CurQueue`. Please use the `CurQueueDevice` variable.